index.bs

```diff
@@ -8083,21 +8079,395 @@ objects.
                 return <emu-val>undefined</emu-val>.
             1.  Otherwise, return the result of performing any steps that were required to be run if the promise was fulfilled,
                 with |value| as the promise’s value.
-    1.  Let |onRejected| be a new [=built-in function object=] whose
-        behavior when invoked is as follows:
+    1.  Let |onFulfilled| be [=!=] [$CreateBuiltinFunction$](|onFulfilledSteps|, « »):
+    1.  Let |onRejectedSteps| be the following steps:
         1.  Let |R| be the first argument to |onRejected|.
         1.  Let |reason| be the result of [=converted to an IDL value|converting=]
             |R| to an IDL value of type {{any}}.
         1.  If there are no steps that are required to be run if the promise was rejected, then
             return <emu-val>undefined</emu-val>.
         1.  Otherwise, return the result of performing any steps that were required to be run if the promise was rejected,
             with |reason| as the rejection reason.
-    1.  Return [=!=] [$PerformPromiseThen$](|promise|, |onFulfilled|, |onRejected|).
+    1.  Let |onRejected| be [=!=] [$CreateBuiltinFunction$](|onRejectedSteps|, « »):
+    1.  Let |constructor| be |promise|.\[[Promise]].\[[Realm]].\[[Intrinsics]].[[{{%Promise%}}]].
+    1.  Let |newCapability| be ? [$NewPromiseCapability$](|constructor|).
+    1.  Return [=!=] [$PerformPromiseThen$](|promise|.\[[Promise]], |onFulfilled|, |onRejected|, |newCapability|).
+
+    Note: If the provided steps return a value of type |T|, this algorithm returns a
+    <a interface lt="Promise">Promise&lt;|T|&gt;</a>.
 </div>
 
-<p class="issue">
-    Include an example of how to write spec text using this term.
-</p>
+<div algorithm>
+    To <dfn export for=Promise lt="a new promise">create</dfn> a new
+    <a interface lt="Promise">Promise&lt;<var ignore>T</var>&gt;</a> in a [=Realm=] |realm|,
+    perform the following steps:
+
+    1.  Let |constructor| be |realm|.\[[Intrinsics]].[[{{%Promise%}}]].
+    1.  Return [=?=] [$NewPromiseCapability$](|constructor|).
+</div>
+
+<div algorithm>
+    To create a <dfn export for=Promise>resolved promise</dfn> of type
```

Comment:
"create a promise resolved with" or "a promise resolved with" is the common usage; it would be good to have those either as the main dfn text or as alternate linking text.